composer/trainer/trainer.py

```diff
@@ -898,6 +916,48 @@ def __init__(
 
         # Load Checkpoint
         self._rng_state = None
+        # If autoresume is enabled, first check for existing checkpoints to load
+        if autoresume:
+            if save_folder is None:
+                raise ValueError("save_folder must be specified when autoresume is enabled.")
+            if save_overwrite:
+                raise ValueError(
+                    "save_overwrite must be False when autoresume is enabled as autoresume always loads the latest existing checkpoint in save_folder."
+                )
+            if save_latest_filename is None:
+                raise ValueError(
+                    "save_latest_filename must be specified so autoresume knows where to load checkpoints from.")
+            if run_name is None:
+                raise ValueError("run_name must be specified so autoresume knows which run to load from.")
+            latest_filename_formatted = format_name_with_dist(save_latest_filename, run_name)
```

Comment:
Following @hanlint's comment above, would it make sense to move this logic (and below) be moved into the `load_checkpoint` function?

We could change the `object_store` parameter of the `load_checkpoint` function to take `Logger | ObjectStore | Sequence[Logger | ObjectStore]`

For the validation (before this line), agree that could be moved into a helper -- e.g. `_validate_autoresume()`

Reply:
Went with the first suggestion, because I think it's better to keep load_checkpoint as a function that just... loads checkpoint lol. More intuitive for a user if they're building something themselves